lib/auth/auth.go

```diff
@@ -3637,6 +3637,10 @@ func (a *Server) getValidatedAccessRequest(ctx context.Context, identity tlsca.I
 		return nil, trace.BadParameter("access request %q has expired", accessRequestID)
 	}
 
+	if req.GetAssumeTime() != nil && req.GetAssumeTime().After(a.GetClock().Now()) {
```

Comment:
Any concerns about time zones here? What if the request is submitted in Europe but the auth server is running in California?

Reply:
That shouldn't matter here I think, since RFC3339 is UTC + some offset. So no matter what the local time zone is the same string should be parsed as the same timestamp.